lib/internal/util/parse_args/parse_args.js

```diff
@@ -0,0 +1,315 @@
+'use strict';
+
+const {
+  ArrayPrototypeForEach,
+  ArrayPrototypePushApply,
+  ArrayPrototypeShift,
+  ArrayPrototypeSlice,
+  ArrayPrototypePush,
+  ArrayPrototypeUnshiftApply,
+  ObjectPrototypeHasOwnProperty: ObjectHasOwn,
+  ObjectEntries,
+  StringPrototypeCharAt,
+  StringPrototypeIncludes,
+  StringPrototypeIndexOf,
+  StringPrototypeSlice,
+} = primordials;
+
+const {
+  validateArray,
+  validateObject,
+  validateString,
+  validateUnion,
+  validateBoolean,
+} = require('internal/validators');
+
+const {
+  findLongOptionForShort,
+  isLoneLongOption,
+  isLoneShortOption,
+  isLongOptionAndValue,
+  isOptionValue,
+  isOptionLikeValue,
+  isShortOptionAndValue,
+  isShortOptionGroup,
+  objectGetOwn,
+  optionsGetOwn,
+} = require('internal/util/parse_args/utils');
+
+const {
+  codes: {
+    ERR_INVALID_ARG_VALUE,
+    ERR_PARSE_ARGS_INVALID_OPTION_VALUE,
+    ERR_PARSE_ARGS_UNKNOWN_OPTION,
+    ERR_PARSE_ARGS_UNEXPECTED_POSITIONAL,
+  },
+} = require('internal/errors');
+
+function getMainArgs() {
+  // This function is a placeholder for proposed process.mainArgs.
+  // Work out where to slice process.argv for user supplied arguments.
+
+  // Electron is an interesting example, with workarounds implemented in
+  // Commander and Yargs. Hopefully Electron would support process.mainArgs
+  // itself and render this workaround moot.
+  //
+  // In a bundled Electron app, the user CLI args directly
+  // follow executable. (No special processing required for unbundled.)
+  // 1) process.versions.electron is either set by electron, or undefined
+  //    see: https://www.electronjs.org/docs/latest/api/process#processdefaultapp-readonly
+  // 2) process.defaultApp is undefined in a bundled Electron app, and set
+  //    in an unbundled Electron app
+  //    see: https://www.electronjs.org/docs/latest/api/process#processdefaultapp-readonly
+  // (Not included in tests as hopefully temporary example.)
+  /* c8 ignore next 3 */
+  if (process.versions?.electron && !process.defaultApp) {
```

Comment:
should this conditional be cached at module level (and evaluated on node startup) so users can’t forge it?

Reply:
Go ahead and submit a patch to `pkgjs/parseargs` with updated tests, if you have an idea of what you're picturing.